### PR TITLE
add eea for string and map method(s)

### PR DESCRIPTION
**Pull request by maggu2810**

### CODE REVIEW

File: libraries/java/java/util/Map.eea [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+containsValue
+ (Ljava/lang/Object;)Z
+ (L0java/lang/Object;)Z
 entrySet
  ()Ljava/util/Set<Ljava/util/Map$Entry<TK;TV;>;>;
  ()Ljava/util/Set<L1java/util/Map$Entry<TK;TV;>;>;
 get
  (Ljava/lang/Object;)TV;
- (Ljava/lang/Object;)T0+V;
+ (L1java/lang/Object;)T0V;
```
Comment (vorburger) [inline libraries/java/java/util/Map.eea]: @maggu2810 do you happen to know what this change means? 

I'm not yet used to reading this syntax.. ;) In and ideal world, we should have https://github.com/lastnpe/eclipse-null-eea-augments/issues/16, one day...

File: libraries/java/java/util/Map.eea [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
 put
  (TK;TV;)TV;
  (TK;TV;)T0V;
+putAll
+ (Ljava/util/Map<+TK;+TV;>;)V
+ (L1java/util/Map<+TK;+TV;>;)V
 remove
  (Ljava/lang/Object;)TV;
- (Ljava/lang/Object;)T0V;
+ (L1java/lang/Object;)T0V;
```
Comment (vorburger) [inline libraries/java/java/util/Map.eea]: and also this, as above, do you happen to know what this changes?

Comment (maggu2810) [reply libraries/java/java/util/Map.eea]:
```
@Nullable T remove(@NonNull Object);
```

It states that the argument must not be null and the return value could be null.

The JavaDoc states:

```
Returns the value to which this map previously associated the key, or null if the map contained no mapping for the key.
```

But it also states:

```
NullPointerException - if the specified key or value is null, and this map does not permit null keys or values (optional)
```

So, a Map itself does not forbid a nullable key (correct?).

I will close this PR. I was not aware that the Map interface allows `null` keys.

Sorry.

Comment (maggu2810) [reply libraries/java/java/util/Map.eea]: I assume the handling should depend on the generic type as this one declare if you allow a nullable key or not.
If you use `Map<@NonNull K, @Nullable V>` you already state that the key of the map is never nullable. WDYT?

Comment (vorburger) [reply libraries/java/java/util/Map.eea]: > So, a Map itself does not forbid a nullable key (correct?).

Yeah I think what that means is basically "it depends on the implementation" - PITA.

> Sorry. I will close this PR. I was not aware that the Map interface allows null keys.

don't be sorry - thanks a lot for taking time to contribute to this at all - community appreciates it!

I thought your other contributions here, the additions apart from the suggested changes, would still be very useful - don't you want to re-open this and just undo the changes instead? Then I'd love the merge it with just the proposed additions! (With a comment like in https://github.com/lastnpe/eclipse-null-eea-augments/pull/23.)

Comment (maggu2810) [reply libraries/java/java/util/Map.eea]: I have already other annotations added in my local copy, so also the topic needs to be changed.
I will create another PR later with a bunch of files.

**Pull request closed without merging**